assets/js/phoenix_live_view/live_socket.js

```diff
@@ -775,10 +775,12 @@ export default class LiveSocket {
     }
     let scroll = window.scrollY
     this.withPageLoading({to: href, kind: "redirect"}, done => {
-      this.replaceMain(href, flash, () => {
-        Browser.pushState(linkState, {type: "redirect", id: this.main.id, scroll: scroll}, href)
-        DOM.dispatchEvent(window, "phx:navigate", {detail: {href, patch: false, pop: false}})
-        this.registerNewLocation(window.location)
+      this.replaceMain(href, flash, (linkRef) => {
+        if(linkRef == this.linkRef) {
+          Browser.pushState(linkState, {type: "redirect", id: this.main.id, scroll: scroll}, href)
```

Comment:
Do we need to worry about possibly losing this scroll position if we don't run this?